src/System.Data.SqlClient/src/System/Data/SqlClient/SqlErrorCollection.cs

```diff
@@ -2,64 +2,36 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-
-
-//------------------------------------------------------------------------------
-
 using System.Collections;
-
+using System.Collections.Generic;
 
 namespace System.Data.SqlClient
 {
     public sealed class SqlErrorCollection : ICollection
     {
-        private ArrayList _errors = new ArrayList();
+        // Ideally this would be typed as List<SqlError>, but that would make the non-generic
+        // CopyTo behave differently than the full framework (which uses ArrayList), throwing
+        // ArgumentException instead of the expected InvalidCastException for incompatible types.
+        // Instead, we use List<object>, which makes the non-generic CopyTo behave like
+        // ArrayList.CopyTo.
+        private readonly List<object> _errors = new List<object>();
 
-        internal SqlErrorCollection()
-        {
-        }
+        internal SqlErrorCollection() { }
 
-        public void CopyTo(Array array, int index)
-        {
-            _errors.CopyTo(array, index);
-        }
+        public void CopyTo(Array array, int index) => ((ICollection)_errors).CopyTo(array, index);
```

Comment:
Feel free to open an separate issue to discuss.

There have definitely been other collections where `CopyTo(Array, int)` was changed to be explicit in .NET Core where it is implicit in the .NET Framework (e.g. `CaptureCollection`: [.NET Core](https://github.com/dotnet/corefx/blob/d228cf36d3768064478baf2af7153d4b0e2aaf5c/src/System.Text.RegularExpressions/ref/System.Text.RegularExpressions.cs#L27), [.NET Framework](http://referencesource.microsoft.com/#System/regex/system/text/regularexpressions/RegexCaptureCollection.cs,2486c4738a8f2987,references)).

However, there are also [cases where it was left as-is to match the full framework surface area](https://github.com/dotnet/corefx/pull/6812#discussion_r55783525), to ease porting/compatibility.